### ISSUE TITLE
v2.0.5 cannot use OpenLDAP when use LDAP BIND with simple authentication method

### ISSUE THREAD

**Issue by kaishuu0123** [rancher/rancher#14452]
**Rancher versions:**
rancher/rancher:latest (v2.0.5)

**Steps to Reproduce:**
* Configure OpenLDAP ( &lt;hostname&gt;/g/security/authentication/openldap )
* Test and enable authentication

**Results:**
* Test and enable authentiation fail.

**DEBUG LOG**

enable debug logging.
* https://rancher.com/docs/rancher/v2.x/en/faq/technical/

```
2018/07/09 04:03:59 [DEBUG] configApplyInput &{{{ } {      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {{ } {      ap[] [] nil [] } openLdapConfig true unrestricted []} [ldap.example.com] 636 true  cn=xxxx,dc=example,dc=com XXXXXX ouinetOrgPerson cn memberOf  ou=groups,dc=example.com cn groupOfNames cn entryDN entryDN member 5000 true} kaishuu0123 XXXXXXXXXXX}
2018/07/09 04:03:59 [DEBUG] Now generating Ldap token
2018/07/09 04:03:59 [DEBUG] Now creating Ldap connection
2018/07/09 04:03:59 [DEBUG] Binding username password
2018/07/09 04:03:59 [ERROR] API error response 401 for POST /v3/openLdapConfigs/openldap?action=testAndApply. Cause: LDAP Result Code 53 "U
2018/07/09 04:03:59 [DEBUG] REQUEST 3893783485235887312 DATA   [25]: buffered
```

**What is the expected result?**
* test login successfully.
* DEBUG output `Bind service account username password`

**Source Code reading**
* Decode from input when testAndApply
    * https://github.com/rancher/rancher/blob/v2.0.5/pkg/auth/providers/ldap/ldap_actions.go#L39
* Detail loginUser
    * https://github.com/rancher/rancher/blob/v2.0.5/pkg/auth/providers/ldap/ldap_client.go#L22
* Detail AuthenticateServiceAccountUser
    * https://github.com/rancher/rancher/blob/v2.0.5/pkg/auth/providers/common/ldap/ldap_util.go#L169

I expected output `Bind service account username password`. but nothing.

* https://github.com/rancher/rancher/blob/v2.0.5/pkg/auth/providers/ldap/ldap_actions.go#L51
    * Decode issue? or Is the request incorrect?

**Comment by superseb** [comment]
What OpenLDAP server is running? Does the given service account have enough permissions to search the tree?

**Comment by kaishuu0123** [comment]
same issue: refs #14456

> What OpenLDAP server is running?

| OS | Ubuntu Trusty (14.04) |
| OpenLDAP server | slapd `2.4.31-1+nmu2ubuntu8.3 amd64` |

* use TLS (port 636)

> Does the given service account have enough permissions to search the tree?

yes.
I use LDAP for basic authentication of my service. But, it works without problems

**Comment by kaishuu0123** [comment]
@superseb 

same issue: #14456

> What OpenLDAP server is running?

|    |   |
|---|---|
| OS | Ubuntu Trusty (14.04) |
| OpenLDAP server | slapd `2.4.31-1+nmu2ubuntu8.3 amd64` |

* use TLS (port 636)

> Does the given service account have enough permissions to search the tree?

yes.
I use LDAP for basic authentication of my service. But, it works without problems

**Comment by kaishuu0123** [comment]
slapd log

```
5b42d396 conn=265493 fd=121 ACCEPT from IP=XXX.XXX.XXX.XXX:33598 (IP=0.0.0.0:636)
5b42d396 conn=265493 fd=121 TLS established tls_ssf=128 ssf=128
5b42d396 conn=265493 op=0 SRCH base="ou=people,dc=example,dc=com" scope=2 deref=0 filter="(&(objectClass=inetOrgPerson)(u
id=kaishuu0123))"
5b42d396 conn=265493 op=0 SRCH attr=dn memberOf objectClass posixAccount uid uid 1.1
5b42d396 conn=265493 op=0 SEARCH RESULT tag=101 err=53 nentries=0 text=authentication required
5b42d396 conn=265493 fd=121 closed (connection lost)
```

**Comment by carolynhu** [comment]
@kaishuu0123 We cannot reproduce your reported issue. Which user do you use to login? Your service account username and password? Since according to the log you posted. The failure seems related to your login user. So I want to know which kind of user are you using to login?

**Comment by kaishuu0123** [comment]
@carolynhu 
* Rancher UI login user
    * admin user (Default Account by setup)
* Service Account username and password
    * I am using an account that can search trees.
    * **Simple Authentication enabled.**
        * https://itconnect.uw.edu/wares/msinf/authn/ldap/ldap-authentication-primer/
* Test and enable authentication
    * Personal Account uid and password

refs: https://github.com/rancher/rancher/pull/14463

**Comment by cjellick** [comment]
We believe we got to the bottom of this issue and why we didn't catch it in QA. Utlimately, our test openldap server was allowing anonymous binds, so it was not erroring out where a typical production openldap server would. We disallowed anonymous binds in our server and where able to reproduce  the problem.

**Comment by deniseschannon** [comment]
@kaishuu0123  We just released v2.0.6-rc1 if you'd like to test it out to see that it's working your openldap servers, that'd be great.

**Comment by deniseschannon** [comment]
Thanks for the update @kaishuu0123!
